sdk/src/types/block/output/feature/block_issuer.rs

```diff
@@ -251,79 +255,70 @@ impl WorkScore for BlockIssuerFeature {
     }
 }
 
+#[derive(From)]
+pub enum BlockIssuerKeySource {
+    ImplicitAccountAddress,
+    PublicKey(PublicKey),
+    Bip44Path(Bip44),
+}
+
 #[cfg(feature = "serde")]
 mod dto {
-    use alloc::{string::String, vec::Vec};
+    use alloc::vec::Vec;
 
     use serde::{Deserialize, Serialize};
 
     use super::*;
-    use crate::types::block::{slot::SlotIndex, Error};
-
-    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize, From)]
-    #[serde(untagged)]
-    pub enum BlockIssuerKeyDto {
-        Ed25519(Ed25519BlockIssuerKeyDto),
-    }
-
-    impl From<&BlockIssuerKey> for BlockIssuerKeyDto {
-        fn from(value: &BlockIssuerKey) -> Self {
-            match value {
-                BlockIssuerKey::Ed25519(s) => Self::Ed25519(s.into()),
-            }
-        }
-    }
-
-    impl TryFrom<BlockIssuerKeyDto> for BlockIssuerKey {
-        type Error = Error;
-
-        fn try_from(value: BlockIssuerKeyDto) -> Result<Self, Self::Error> {
-            match value {
-                BlockIssuerKeyDto::Ed25519(s) => Ok(Self::Ed25519(s.try_into()?)),
-            }
-        }
-    }
+    use crate::{
+        types::block::{slot::SlotIndex, Error},
+        utils::serde::prefix_hex_bytes,
+    };
 
     #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
     #[serde(rename_all = "camelCase")]
-    pub struct Ed25519BlockIssuerKeyDto {
+    pub struct Ed25519PublicKeyHashBlockIssuerKeyDto {
         #[serde(rename = "type")]
         pub kind: u8,
-        pub public_key: String,
+        #[serde(with = "prefix_hex_bytes")]
+        pub pub_key_hash: [u8; Ed25519PublicKeyHashBlockIssuerKey::LENGTH],
```

Comment:
could just use the `Ed25519PublicKeyHashBlockIssuerKey` type directly, right?

Reply:
If we make it has a `pub_key_hash` field, yeah. Want me to do it ?

Reply:
no I mean can't you just do this?

```rust
#[serde(with = "prefix_hex_bytes")]
pub pub_key_hash: Ed25519PublicKeyHashBlockIssuerKey,
```

Reply:
You want the DTO to use the real type to serialize?

Reply:
yeah, why not?

Reply:
Isn't that sort of cyclic?